bazel/cert.bzl

```diff
@@ -0,0 +1,130 @@
+
+    Args:
+      name: name of the target
+      certificate: name to use for output files (crt, key, and csr)
+      common_name: the CN to use when setting the subject name
+    """
+
+    cert = certificate + ".crt"
+    subj = "/C=US/ST=California/L=San Francisco/O=Redpanda Data/OU=Core/CN=" + common_name
+
+    pr_key_gen = _redpanda_private_key(name, certificate)
+
+    crt_gen = name + "_crt_gen"
+    native.genrule(
+        name = crt_gen,
+        srcs = [
+            pr_key_gen,
+        ],
+        outs = [cert],
+        cmd = "$(execpath @openssl//:openssl_exe) req " +
+              "-new -x509 -sha256 " +
+              "-key $(SRCS) " +
+              "-out \"$@\" " +
+              "-subj \"{}\" ".format(subj) +
+              "-addext \"subjectAltName = IP:127.0.0.1\"",
+        tools = [
+            "@openssl//:openssl_exe",
+        ],
+    )
+
+    native.filegroup(
+        name = name,
+        srcs = [pr_key_gen, crt_gen],
+        visibility = [
+            "//visibility:public",
+        ],
+    )
+
+def redpanda_signed_cert(name, certificate, common_name, ca, serial_number):
+    """
+    Generate a Redpanda signed certificate.
+
+    Args:
+      name: name of the target
+      certificate: name to use for output files (crt, key, and csr)
+      common_name: the CN to use when setting the subject name
+      ca: the certificate to be used as the signing CA
+      serial_number: the serial number of cert when issued by CA
+    """
+
+    subj = "/C=US/ST=California/L=San Francisco/O=Redpanda Data/OU=Core/CN=" + common_name
+
+    pr_key_gen = _redpanda_private_key(name, certificate)
+
+    req_gen = name + "_csr_gen"
+    native.genrule(
+        name = req_gen,
+        srcs = [
+            pr_key_gen,
+        ],
+        outs = [certificate + ".csr"],
+        cmd = "$(execpath @openssl//:openssl_exe) req " +
+              "-new -sha256 " +
+              "-key $(SRCS) " +
+              "-out \"$@\" " +
+              "-subj \"{}\" ".format(subj),
+        tools = [
+            "@openssl//:openssl_exe",
+        ],
+    )
+
+    ca_cert = ca + ".crt"
+    ca_private_key = ca + ".key"
+
+    crt_gen = name + "_crt_gen"
+    native.genrule(
+        name = crt_gen,
+        srcs = [
+            ca_cert,
+            ca_private_key,
+            req_gen,
+        ],
+        outs = [certificate + ".crt"],
+        cmd = "$(execpath @openssl//:openssl_exe) x509 " +
+              "-req -days 1000 -sha256 " +
+              "-set_serial {} ".format(serial_number) +
+              "-in $(execpath {}) ".format(req_gen) +
+              "-CA $(execpaths :{}) ".format(ca_cert) +
+              "-CAkey $(execpaths :{}) ".format(ca_private_key) +
+              "-out \"$@\" ",
+        tools = [
+            "@openssl//:openssl_exe",
+        ],
+    )
+
+    native.filegroup(
+        name = name,
+        srcs = [pr_key_gen, req_gen, crt_gen],
+        visibility = [
+            "//visibility:public",
```

Comment:
This visibility should be like other rules and be what the user passes in and defaulting to None